Docs: Update basic roles permissions using curl (#90485)
grafana/grafana: docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md

By default, only a Grafana Server Admin can create and manage custom roles. If you want your `Editors` to do the same, [update the `Editor` basic role permissions]({{< ref "./manage-rbac-roles.md#update-basic-role-permissions" >}}). There are two ways to achieve this:

- Add the following permissions to the `basic:editor` role, using provisioning or the [RBAC HTTP API]({{< relref "../../../../developers/http_api/access_control/#update-a-role" >}}):

  | action         | scope                       |
  | -------------- | --------------------------- |
  | `roles:read`   | `roles:*`                   |
  | `roles:write`  | `permissions:type:delegate` |
  | `roles:delete` | `permissions:type:delegate` |

>   As an example, here is a small bash script that fetches the role, modifies it using `jq` and updates it:

Reviewer: ```suggestion
  As an example, here is a small Bash script that fetches the role, modifies it using `jq` and updates it:
```

  ```bash

>   # Fetch the role, modify it to add the desired permissions and increment its version

Reviewer: Generally it's better to provide this information outside of the code block so that it can be more easily read by the user.

  curl -H 'Authorization: Bearer glsa_kcVxDhZtu5ISOZIEt' \
    -X GET '<grafana_url>/api/access-control/roles/basic_editor' | \
    jq 'del(.created)| del(.updated) | del(.permissions[].created) | del(.permissions[].updated) | .version += 1' | \

>     jq '.permissions += [{"action": "roles:read", "scope": "roles:*"}, {"action": "roles:write", "scope": "permissions:type:delegate"}, {"action": "roles:delete", "scope": "permissions:type:delegate"}]' > /tmp/basic_editor.json

Reviewer: Placeholders should use the style documented in https://grafana.com/docs/writers-toolkit/write/style-guide/write-for-developers/#placeholder-variables.

It seems that the Bearer token should also be a placeholder variable.

I think given the complexity of the script, it's worth explaining the placeholder variables and also splitting each individual command into its own code block since they can't be copy-pasted without modification anyway.

Then the code comments could become the numbered steps.

  # Update the role
  curl -H 'Authorization: Bearer glsa_kcVxDhZtu5ISOZIEt' -H 'Content-Type: application/json' \
    -X PUT-d @/tmp/basic_editor.json '<grafana_url>/api/access-control/roles/basic_editor'
  ```

> - Or add the `fixed:roles:writer` role permissions to the `basic:editor` role using the `role > from` list of your provisioning file:

Reviewer: The key in the YAML snippet is `roles` not `role`.

We encourage that you use dot syntax to refer to nested fields: https://grafana.com/docs/writers-toolkit/write/style-guide/write-for-developers/#refer-to-nested-field so it would be something like:

```suggestion
- You can instead add the `fixed:roles:writer` role permission to the `basic:editor` role by extending the `roles.from` list in your provisioning file:
```
